ninjareact/Dockerfile

```diff
@@ -0,0 +1,47 @@
+# Use an official Python runtime as a parent image
+FROM python:3.11-slim
+
+# Set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+ENV DJANGO_ENV=production
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    libpq-dev \
+    curl \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Node.js and npm for building the React frontend
+RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash - \
+    && apt-get install -y nodejs
+
+# Copy the requirements file into the container
+COPY requirements.txt /app/
+
+# Install Python dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the rest of the application code into the container
+COPY . /app/
+
+# Build the React frontend
+WORKDIR /app/frontend
+RUN npm install && npm run build
+
+# Collect static files (including React build)
+WORKDIR /app
+RUN python manage.py collectstatic --noinput
+
+# Run database migrations
+RUN python manage.py migrate
```

Comment:
As written, this creates a `db.sqlite3` file in the image. This should be done at runtime, because in production a "real" database service will be used.